frontend/src/components/FieldOfLawNodeComponent.vue

```diff
@@ -1,178 +1,186 @@
 <script lang="ts" setup>
-import { computed, onMounted, toRaw } from "vue"
+import { computed, ref, watch } from "vue"
+import { NodeHelperInterface } from "@/components/fieldOfLawNode"
+import FlexContainer from "@/components/FlexContainer.vue"
 import Checkbox from "@/components/input/CheckboxInput.vue"
 import TokenizeText from "@/components/TokenizeText.vue"
-import { ROOT_ID, FieldOfLawNode } from "@/domain/fieldOfLaw"
-import FieldOfLawService from "@/services/fieldOfLawService"
+import { FieldOfLaw } from "@/domain/fieldOfLaw"
 import IconAdd from "~icons/ic/baseline-add"
 import IconHorizontalRule from "~icons/ic/baseline-horizontal-rule"
 
 interface Props {
-  selectedNodes: FieldOfLawNode[]
-  node: FieldOfLawNode
-  selected: boolean
+  node: FieldOfLaw
+  modelValue: FieldOfLaw[]
   showNorms: boolean
+  nodeHelper: NodeHelperInterface
+  expandValues: FieldOfLaw[]
+  isRoot?: boolean
+  selectedNode?: FieldOfLaw
 }
 
 const props = defineProps<Props>()
+
 const emit = defineEmits<{
-  "node:select": [node: FieldOfLawNode]
-  "node:unselect": [node: FieldOfLawNode]
-  "linkedField:clicked": [identifier: string]
+  "node:select": [node: FieldOfLaw]
```

Comment:
[Non-blocking]
This again just wishful thinking of a bigger refactoring of this component: 
Can't we just emit a `selected` boolean value? How does the selected value differ from linkedField:clicked? Do we actually need to emit expand and collapse, because we can derive the information also from the `selected` value?

Reply:
One needs to know the selected node. The selected node is that object that came from either from the search or from the other list. The tree needs to know which node to show the direct path to. The boolean value is not enough. `linkedField` is a mystery for me. I don't know why, anyway, we had a linked field that is differed from node fieldOfLaw. In my offer I created a node out of this linkedField but did not want to get rid of it completely as now I know clearer and clearer that is just not needed in code.

Reply:
💡Ok good to know!